Dockerfile

```diff
@@ -1,72 +1,109 @@
-# Build Python app.
-FROM python:3.8-bullseye AS backend
+# We use Debian images because they are considered more stable than the alpine
+# ones becase they use a different C compiler. Debian images also come with
+# all useful packages required for image manipulation out of the box. They
+# however weight a lot, approx. up to 1.5GiB per built image.
+FROM python:3.8 as production
 
-ENV PYTHONUNBUFFERED=1 \
-    PYTHONPATH=/app \
-    DJANGO_SETTINGS_MODULE=wagtailio.settings.production \
-    PORT=8000 \
-    WEB_CONCURRENCY=3 \
-    GUNICORN_CMD_ARGS="--max-requests 1200 --access-logfile -"
+ARG POETRY_HOME=/opt/poetry
+ARG POETRY_INSTALL_ARGS="--no-dev"
 
-# Install operating system dependencies.
-RUN apt-get update -y && \
-    apt-get install -y apt-transport-https rsync libmagickwand-dev unzip postgresql-client-13 && \
-    rm -rf /var/lib/apt/lists/*
+# IMPORTANT: Remember to review both of these when upgrading
+ARG POETRY_VERSION=1.1.13
+# To get this value locally:
+# $ wget https://raw.githubusercontent.com/python-poetry/poetry/1.1.13/get-poetry.py
+# $ sha1sum get-poetry.py
+ARG POETRY_INSTALLER_SHA=eedf0fe5a31e5bb899efa581cbe4df59af02ea5f
 
-WORKDIR /app
-EXPOSE 8000
+# Install dependencies in a virtualenv
+ENV VIRTUAL_ENV=/venv
 
-# Create a non-root application user.
-ARG UID=1000
-RUN useradd wagtailio -u $UID -m
```

Comment:
Please could we keep the ``UID`` argument? This makes the development experience a bit nicer as any files that are created by Django would be given the same UID as the host user so there won't be any permissions issues.